### ISSUE TITLE
Add solar equator and Laplace invariable planes to Celestial Sphere

### ISSUE THREAD

**Issue by alex-w** [Stellarium/stellarium#358]
_Original report:_ https://bugs.launchpad.net/stellarium/+bug/1639336

Converted from blueprint:

Request addition of a line in constellation settings to represent the plane of the solar system (not the ecliptic), which is a line projected on the heavens through the solar equator. In addition, request a similar line drawn to reflect the barycenter (center of mass) of the solar plane, also known as the Laplace plane (see wiki). Both are used by amateur astronomers searching for undiscovered orbiting bodies, as the assumption is often that the majority of orbiting bodies in our solar system lie near these planes. My apologies if I made this request incorrectly.
http://en.wikipedia.org/wiki/Invariable_plane

**Comment by alex-w** [comment]
The solar equator already added in ArchaeoLines plugin

**Comment by gzotti** [comment]
Alex, I am afraid no. The solar equator is a bit off from the ecliptic plane (which is the relevant plane in cultural astronomy and ArchaeoLines) and is not relevant to astronomy before around the 19th century.

**Comment by alex-w** [comment]
The solar equator is called "Current Sun" in the plugin

**Comment by gzotti** [comment]
no, this is the daily track of the sun (just its current declination!) There is also "Current Moon".

**Comment by alex-w** [comment]
Yes, of course. Definition: The solar equator is the latitude on Earth at which the Sun is directly overhead; where the sun is vertically above at midday.

**Comment by gzotti** [comment]
Wel, OP wanted this as plane of the real Solar Equator. Nothing to do with earth. Yours I would call solar zenith passage.

**Comment by alex-w** [comment]
Maybe not :) I've asked Roger for comment of his proposal.

**Comment by rmorrison0324** [comment]
Howdy all.  Rodger here.  Sorry for the delay.  I had trouble getting set in GitHub.

About my request.  I'm an astrophotographer and I am beginning to get involved in comet and asteroid hunting.  The Invariable Plane (not the Laplace Plane, which I said in error in the original post) is, essentially, that plane that contains the average of the orbital planes of all objects in the solar system, weighted by mass.  Most planets and asteroids orbit close to this plane, so it makes sense to keep searches for such near that plane, statistically speaking.  Visualizing what this would look like, I would think, would be much like how a planet ring would look if observed from inside a ring's orbit (only with the sun).  The solar equator would be only slightly different.  Either way, to be useful, the plane would have to show not only a location X-Y, but would also need to involve a distance (z), which would be nearly impossible on a 2-dimensional screen.  Imagine if you were on Saturn and its rings were infinitely large.  They would probably take up most of the screen.  The only thing that would make any sense would be to draw a line through the cosmos that represents the infinite edge of the plane.  This would be nice to know, but would not really be as usable for my purposes.  I suspect that I would be a "one-off" user and it would not be a good feature addition to Stellarium.

That said, I now understand how it will likely be a nightmare to calculate the location of the planes themselves, let alone how one would draw them in Stellarium.  They can't be static.  The have their own precession patterns.

So, in short, I withdraw my request.  I will just trace the current location of the planes on a paper star chart and work from that.

Kindest regards (and with much thanks for your attention).

Rodger

p.s. my apologies for having wasted your time.

**Comment by gzotti** [comment]
Thanks for clarifying @rmorrison0324 . Indeed the solution is quite ambitious. We can leave this idea still open, maybe somebody wants to implement it.

**Comment by gzotti** [comment]
@rmorrison0324 @alex-w The orientation of the Invariable Plane of the solar system is given in the IAU WGCCRE reports. I could add this as one more Great Circle in the GridlineMgr if there is any use for it. 
Likewise, the projection of the Solar equator out into space could be visualized with another Great Circle. One of the unsolved questions (AFAIK) is why the solar equator is not parallel to the ecliptic.

**Comment by alex-w** [comment]
Please do it :)

**Comment by axd1967** [comment]
Would a workaround, if you know the location of this invariable plane and need to plot it anyway, be to use the measuring tool to approximate a section of it?

![image](https://user-images.githubusercontent.com/3529789/103777727-9ba78a00-5031-11eb-92d2-160d623f8710.png)

**Comment by gzotti** [comment]
The easiest workaround is showing Ecliptic of J2000. Just that it is *not* exactly that. Already testing my solution...

**Comment by Atque** [comment]
@gzotti Would poles of the invariable plane and poles of the Sun's rotation axis projected into the sky make sense?

**Comment by gzotti** [comment]
I return the question to you. Is there any practical application for those?

**Comment by Atque** [comment]
@gzotti I don't think there is, but I'm not too much into hunting undiscovered bodies.

**Comment by gzotti** [comment]
The request was for the planes, not poles. The Invariable Plane lies very close to the ecliptic pole, which can act as guide for the common "rough estimates" e.g. where zodiacal light disturbs the least. If anybody has a particular interest with good explanation why Ecliptic Pole is not enough, these points can be added at a later date.
